### PR TITLE
remove basemap definitively

### PR DESCRIPTION
**Pull request by ThomasLecocq**
### What does this PR do?

removing basemap support

### Why was it initiated?  Any relevant Issues?

need to get a clean codebase, and now cartopy plots are available for all plots, so useless to keep both.
#2680 
#2960

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the "build_docs" tag to this PR.
- [ ] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the "test_network" tag to this PR.
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are covered via new tests.
- [ ] Any new or changed features are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [ ] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [ ] Add the "ready for review" tag when you are ready for the PR to be reviewed.

### CODE REVIEW

**Comment by ThomasLecocq** [comment]
updated documentation: http://msnoise.org/obspy-doc/index.html

**Comment by ThomasLecocq** [comment]
note @d-chambers & @trichter running on my machine, the ``test_inventory.class TestInventoryCartopy:`` raises an error for pytest.. Maybe we could add the cartopy package on the default non-skipped tests (at least for 1 platform/py version)?

**Comment by d-chambers** [comment]
Would it work to just add cartopy to the conda envs so that none of the cartopy tests are skipped?

**Comment by ThomasLecocq** [comment]
Yes we can maybe try on this PR

**Comment by ThomasLecocq** [comment]
not sure anyone ever used `plot_basemap` public func directly (or with `plot_map(method='basemap')`, but do you want to keep it to warn about the deprecation & pass the args+kwargs to plot_cartopy ?

**Review by d-chambers — COMMENTED** [review]
Looks like there are still some basemap remnants in the docker/debian files, but perhaps we address those in a different PR? (Honestly I don't know if we are even using the docker files for anything now...)

Is this still needed? https://github.com/obspy/obspy/blob/a29f7d87aa9de910fa4da6ae757cb77a3b6bdddb/obspy/core/util/base.py#L408

Can we remove this docstring or is it still relevant?

https://github.com/obspy/obspy/blob/a29f7d87aa9de910fa4da6ae757cb77a3b6bdddb/obspy/core/util/base.py#L368-L371

Also in this docs file:

https://github.com/obspy/obspy/blob/a29f7d87aa9de910fa4da6ae757cb77a3b6bdddb/misc/docs/py3-docs-master.yml#L13

**Comment by ThomasLecocq** [comment]
@d-chambers @trichter could you have a look at the cartopy tests? I can't figure out why they fail on my machine (cartopy installed, so test should run, but seems pytest fails at "collecting" them)

File: obspy/core/tests/test_inventory.py [hunk 1]
```diff
@@ -583,10 +581,10 @@ def test_read_invalid_filename(self):
         # try read_inventory() with invalid filename for all registered read
         # plugins and also for filetype autodiscovery
         formats = [None] + list(formats)
+        expected_error_message = exception_msg.format(doesnt_exist)
         for format in formats[:1]:
-            with pytest.raises(IOError) as e:
+            with pytest.raises(IOError, match=expected_error_message):
```
Comment (trichter) [inline obspy/core/tests/test_inventory.py:586]: This change introduced the new failure

Either try  `with pytest.raises(IOError, match=re.escape(expected_error_message)):` or revert the change or just check for `IOError` or `FileNotFoundError` without checking the text explicitely.

Comment (d-chambers) [reply obspy/core/tests/test_inventory.py:586]: No, 'pytest.importorskip' was skipping all the tests in this file before, I don't think the test ever passed. I will try to look at later today.

**Comment by d-chambers** [comment]
Looks like I need to merge/rebase. Side note: we need to purge apveyor from our PR checks.

**Comment by d-chambers** [comment]
I think we are good to go but I am leaving the merging to @ThomasLecocq to make sure there isn't anything else to do here.

**Comment by ThomasLecocq** [comment]
all good for me, thanks for the last sprint @d-chambers

**Pull request merged**